checkov/terraform/graph_builder/local_graph.py

```diff
@@ -89,9 +90,34 @@ def build_graph(self, render_variables: bool) -> None:
                 edges_count = len(self.edges)
                 self._build_cross_variable_edges()
                 logging.info(f"Found {len(self.edges) - edges_count} cross variable edges")
+            self._remove_empty_foreach_edges()
         else:
             self.update_vertices_fields()
 
+    def _remove_empty_foreach_edges(self) -> None:
+        """
+        Check for resources which are marked with empty `count/foreach` statements, and remove their matching edges.
+        This makes sure we won't find connections for resources which are not planned to be created.
+        """
+        relevant_indices = [index for list_of_indices in self.foreach_blocks.values() for index in list_of_indices]
+        for foreach_vertex_index in relevant_indices:
+            vertex = self.vertices[foreach_vertex_index]
+            is_planned_to_be_empty_count = vertex.attributes.get("count", [])[0] == 0 \
+                if "count" in vertex.attributes and len(vertex.attributes.get("count", [])) == 1 else False
+            is_planned_to_be_empty_foreach = vertex.attributes.get("for_each") in [{}, []] \
+                if "for_each" in vertex.attributes else False
+            if is_planned_to_be_empty_count or is_planned_to_be_empty_foreach:
+                copied_edges = copy.copy(self.edges)
```

Comment:
What is the performance impact of doing this copy inside the loop? AFAIK we can have a very large amount of edges.
Is it possible to do some pre-processing and alleviate the pressure from the inline processing?
Something like run once on all edges and map out the needed attributes, and then inside the for-loop just search in O(1).